python/pyspark/sql/__init__.py

```diff
@@ -31,9 +31,22 @@
     - L{HiveContext}
       Main entry point for accessing data stored in Apache Hive..
 """
+from __future__ import absolute_import
+
+# fix the module name conflict for Python 3+
```

Comment:
Ah, I think I remember the issue: I think that Python places the main file's enclosing directory onto the Python path to enable relative imports, so I guess that `python/pyspark/sql` ends up on the PYTHONPATH when we directly execute its files in run-tests.  We had a sort of messy hack for dealing with this earlier, but the file rename here also seems fine so let's go ahead and keep it.